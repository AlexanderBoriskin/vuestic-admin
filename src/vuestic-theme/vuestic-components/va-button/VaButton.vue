<template>
  <component
    :is="computedTag"
    class="va-button"
    :class="buttonClass"
    :style="buttonStyle"
    :disabled="disabled"
    :type="type"
    :href="href"
    :target="target"
    :to="to"
    :replace="replace"
    :append="append"
    :active-class="activeClass"
    :exact="exact"
    :exact-active-class="exactActiveClass"
    v-on="inputListeners"
    @mouseenter="updateHoverState(true)"
    @mouseleave="updateHoverState(false)"
    @focus="updateFocusState(true)"
    @blur="updateFocusState(false)"
    tabindex="0"
  >
    <div class="va-button__content">
      <va-icon
        v-if="icon"
        fixed-width
        class="va-button__content__icon va-button__content__icon-left"
        :icon="icon"
      />
      <div
        v-if="hasTitleData"
        class="va-button__content__title">
        <slot/>
      </div>
      <va-icon
        v-if="iconRight"
        fixed-width
        class="va-button__content__icon va-button__content__icon-right"
        :icon="iconRight"
      />
    </div>
  </component>
</template>

<script>
import { getGradientColor, getHoverColor, getFocusColor, getBoxShadowColor } from '../../../services/colors'
import VaIcon from '../va-icon/VaIcon'

export default {
  name: 'va-button',
  components: { VaIcon },
  props: {
    tag: {
      type: String,
      default: 'button'
    },
    outline: {
      type: Boolean
    },
    flat: {
      type: Boolean
    },
    color: {
      type: String,
      default: 'success'
    },
    small: {
      type: Boolean
    },
    large: {
      type: Boolean
    },
    icon: {
      type: String
    },
    iconRight: {
      type: String
    },
    type: {
      type: String
    },
    disabled: {
      type: Boolean
    },
    /* Link props */
    href: {
      type: String
    },
    target: {
      type: String,
    },
    /* Router link props */
    to: {
      type: [String, Object]
    },
    replace: {
      type: Boolean
    },
    append: {
      type: Boolean
    },
    activeClass: {
      type: String
    },
    exact: {
      type: Boolean
    },
    exactActiveClass: {
      type: String
    }
  },
  data () {
    return {
      hoverState: false,
      focusState: false,
    }
  },
  computed: {
    buttonClass () {
      return {
        'va-button--success': this.color === 'success',
        'va-button--info': this.color === 'info',
        'va-button--danger': this.color === 'danger',
        'va-button--warning': this.color === 'warning',
        'va-button--gray': this.color === 'gray',
        'va-button--dark': this.color === 'dark',
        'va-button--default': !this.flat && !this.outline,
        'va-button--flat': this.flat,
        'va-button--outline': this.outline,
        'va-button--disabled': this.disabled,
        'va-button--without-title': !this.hasTitleData,
        'va-button--with-left-icon': this.icon,
        'va-button--with-right-icon': this.iconRight,
        'va-button--large': this.large,
        'va-button--small': this.small,
        'va-button--normal': !this.large && !this.small
      }
    },
    buttonStyle () {
      if (this.focusState) {
        if (this.flat){
          return {
            color: this.themes[this.color],
            background: getFocusColor(this.color)
          }
        }
        if (this.outline){
          return {
            color: this.themes[this.color],
            borderColor: this.themes[this.color],
            background: getFocusColor(this.color)
          }
        } else {
          return {
            backgroundImage: !this.flat && !this.outline
              ? 'linear-gradient(to right,' + getGradientColor(this.color)[0] +
              ',' + getGradientColor(this.color)[1] + ')' : '',
          }
        }
      } else if (this.hoverState) {
        if (this.flat){
          return {
            color: this.themes[this.color],
            background: getHoverColor(this.color)
          }
        }
        if (this.outline){
          return {
            color: this.themes[this.color],
            borderColor: this.themes[this.color],
            background: getHoverColor(this.color)
          }
        } else {
          return {
            backgroundImage: !this.flat && !this.outline
              ? 'linear-gradient(to right,' + getGradientColor(this.color)[0] +
              ',' + getGradientColor(this.color)[1] + ')' : '',
          }
        }
      } else {
        return {
          color: this.flat || this.outline ? this.themes[this.color] : '#ffffff',
          borderColor: this.outline ? this.themes[this.color] : '',
          backgroundImage: !this.flat && !this.outline
            ? 'linear-gradient(to right,' + getGradientColor(this.color)[0] +
              ',' + getGradientColor(this.color)[1] + ')' : '',
          boxShadow: !this.flat && !this.outline ? '0 0.125rem 0.19rem 0 ' + getBoxShadowColor(this.color) : ''
        }
      }
    },
    hasTitleData () {
      return this.$slots.default
    },
    computedTag () {
      if (this.tag === 'a' || this.href || this.target) {
        return 'a'
      }
      if (this.tag === 'router-link' || this.to || this.append || this.replace ||
        this.activeClass || this.exact || this.exactActiveClass) {
        return 'router-link'
      }
      return 'button'
    },
    inputListeners () {
      const vm = this
      return Object.assign({},
        this.$listeners,
        {
          click: function (event) {
            vm.$emit('click', event)
          },
        }
      )
    }
  },
  methods: {
    updateHoverState (isHover) {
      this.hoverState = isHover
    },
    updateFocusState (isHover) {
      this.focusState = isHover
    },
  }
}
</script>

<style lang='scss'>
  @mixin button-size($padding-y, $padding-x, $font-size, $line-height, $border-radius) {
    padding: $padding-y $padding-x;
    font-size: $font-size;
    line-height: $line-height;
    border-radius: $border-radius;
  }

  .va-button {
    display: inline-block;
    margin: $btn-margin;
    background-image: none;
    box-shadow: none;
    outline: none !important;
    border: $btn-border;
    font-family: $font-family-sans-serif;
    text-decoration: none !important;
    text-transform: initial;
    cursor: pointer;
    transition: $btn-transition;

    &__content {
      display: flex;

      &__title, &__icon {
        display: flex;
        justify-content: center;
        align-items: center;
        margin: auto;
      }
    }

    &--default {
      color: $white;

      &:hover {
        filter: brightness(115%);
      }

      &:focus, &:active {
        filter: brightness(85%);
      }

      i {
        color: $white;
      }
    }

    &--outline {
      background-color: transparent;
      border: solid $btn-border-outline;
      text-decoration: none;

      &.va-button--disabled {
        background: transparent;
        @include va-disabled;

        &.va-button--active {
          background-color: $brand-secondary !important;

          .va-button__content, i {
            color: $white !important;
          }
        }

        .va-pagination & {
          border-color: $brand-secondary;
          opacity: 1;

          .va-button__content, i {
            color: $brand-secondary;
          }

          &.va-button--no-effects {

            .va-button__content, i {
              color: $brand-secondary !important;
            }
          }
        }
      }
    }

    &--flat {
      background: transparent;
      border: $btn-border solid transparent;
      text-decoration: none;
    }

    &.va-button--disabled {
      @include va-disabled;
    }

    &--large {
      @include button-size($btn-padding-y-lg, $btn-padding-x-lg, $btn-font-size-lg, $btn-line-height-lg, $btn-border-radius-lg);
      letter-spacing: $btn-letter-spacing-lg;

      .va-button__content__icon {
        width: $btn-icon-width-lg;
      }

      &.va-button--with-left-icon {
        padding-left: $btn-with-icon-wrapper-padding-lg;

        &.va-button--without-title {
          padding-right: $btn-with-icon-wrapper-padding-lg;
        }

        .va-button__content__title {
          padding-left: $btn-with-icon-content-padding-lg;
        }
      }

      &.va-button--with-right-icon {
        padding-right: $btn-with-icon-wrapper-padding-lg;

        .va-button__content__title {
          padding-right: $btn-with-icon-content-padding-lg;
        }
      }
    }

    &--small {
      @include button-size($btn-padding-y-sm, $btn-padding-x-sm, $btn-font-size-sm, $btn-line-height-sm, $btn-border-radius-sm);
      letter-spacing: $btn-letter-spacing-sm;

      .va-button__content__icon {
        width: $btn-icon-width-sm;
      }

      &.va-button--with-left-icon {
        padding-left: $btn-with-icon-wrapper-padding-sm;

        &.va-button--without-title {
          padding-right: $btn-with-icon-wrapper-padding-sm;
        }

        .va-button__content__title {
          padding-left: $btn-with-icon-content-padding-sm;
        }
      }

      &.va-button--with-right-icon {
        padding-right: $btn-with-icon-wrapper-padding-sm;

        .va-button__content__title {
          padding-right: $btn-with-icon-content-padding-sm;
        }
      }
    }

    &--normal {
      @include button-size($btn-padding-y-nrm, $btn-padding-x-nrm, $btn-font-size-nrm, $btn-line-height-nrm, $btn-border-radius-nrm);
      letter-spacing: $btn-letter-spacing-nrm;

      .va-button__content__icon {
        width: $btn-icon-width-nrm;
      }

      &.va-button--with-left-icon {
        padding-left: $btn-with-icon-wrapper-padding-nrm;

        &.va-button--without-title {
          padding-right: $btn-with-icon-wrapper-padding-nrm;
        }

        .va-button__content__title {
          padding-left: $btn-with-icon-content-padding-nrm;
        }
      }

      &.va-button--with-right-icon {
        padding-right: $btn-with-icon-wrapper-padding-nrm;

        .va-button__content__title {
          padding-right: $btn-with-icon-content-padding-nrm;
        }
      }
    }
  }

  $vuestic-colors: (
    success: (#63e5b3, #23e066, #77cea4, #40e583, #d6ffd3, #c0fbc7),
    danger: (#ff7455, #e34b4a, #b86e6d, #e34b4a, #ffebeb, #fbd2d2),
    warning: (#ffd72d, #feb900, #cbb06e, #ffc202, #fff3d1, #ffebb1),
    info: (#32b5e4, #2c82e0, #6c97ac, #2c82e0, #caeeff, #b2defb),
    gray: (#cdd0d5, #b4b6b9, #a3aab0, #babfc2, #e6e9ec, #dfe3e5),
    dark: (#576675, #34495e, #aebcca, #34495e, #afb6bb, #afb6bb)
  );

  @each $name, $colors in $vuestic-colors {
    $gradient-color1: nth($colors, 1);
    $gradient-color2: nth($colors, 2);
    $box-shadow: nth($colors, 3);
    $border-color: nth($colors, 4);
    $hover-color: nth($colors, 5);
    $focus-color: nth($colors, 6);

    .va-button--#{$name}.va-button--outline{

      .va-pagination & {

        &:focus {
          background-color: transparent;

          .va-button__content, i {
            color: $border-color;
          }
        }

        &.va-button--active {
          background-color: $border-color;

          .va-button__content, i {
            color: $white;
          }
        }

        .va-button__content {
          min-width: 20px;
        }

        &.va-button--no-effects {

          &:hover, &:active, &:focus {
            background-color: transparent;

            .va-button__content, i {
              color: $border-color;
            }
          }
        }
      }

      &:active, &:focus, &.va-button--active {

        .va-button-toggle & {
          background-color: $border-color;

          .va-button__content, i {
            color: $white;
          }
        }
      }
    }
  }
</style>
