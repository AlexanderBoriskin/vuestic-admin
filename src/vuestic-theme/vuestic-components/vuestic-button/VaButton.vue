<template>
  <component
    :is="computedTag"
    class="va-button"
    :class="buttonClass"
    :disabled="disabled"
    :type="type"
    :href="href"
    :target="target"
    :to="to"
    :replace="replace"
    :append="append"
    :active-class="activeClass"
    :exact="exact"
    :exact-active-class="exactActiveClass"
  >
    <va-button-content
      :icon="icon"
      :icon-right="iconRight"
    >
      <slot/>
    </va-button-content>
  </component>
</template>

<script>
import VaButtonContent from './VaButtonContent'

export default {
  name: 'va-button',
  components: {
    VaButtonContent
  },
  props: {
    tag: {
      type: String,
      default: 'button'
    },
    outline: {
      type: Boolean
    },
    flat: {
      type: Boolean
    },
    color: {
      type: String,
      default: 'success'
    },
    small: {
      type: Boolean
    },
    large: {
      type: Boolean
    },
    icon: {
      type: String
    },
    iconRight: {
      type: String
    },
    type: {
      type: String
    },
    disabled: {
      type: Boolean
    },
    /* Link props */
    href: {
      type: [String, Object]
    },
    target: {
      type: String,
    },
    /* Router link props */
    to: {
      type: [String, Object]
    },
    replace: {
      type: Boolean
    },
    append: {
      type: Boolean
    },
    activeClass: {
      type: String
    },
    exact: {
      type: Boolean
    },
    exactActiveClass: {
      type: String
    }
  },
  computed: {
    buttonClass () {
      return {
        'va-button--success': this.color === 'success',
        'va-button--info': this.color === 'info',
        'va-button--danger': this.color === 'danger',
        'va-button--warning': this.color === 'warning',
        'va-button--gray': this.color === 'gray',
        'va-button--dark': this.color === 'dark',
        'va-button--default': !this.flat && !this.outline,
        'va-button--flat': this.flat,
        'va-button--outline': this.outline,
        'va-button--disabled': this.disabled,
        'va-button--with-left-icon': this.icon,
        'va-button--with-right-icon': this.iconRight,
        'va-button--large': this.large,
        'va-button--small': this.small,
        'va-button--normal': !this.large && !this.small
      }
    },
    computedTag () {
      if (this.tag === 'a' || this.href || this.target) {
        return 'a'
      }
      if (this.tag === 'router-link' || this.to || this.append || this.replace ||
        this.activeClass || this.exact || this.exactActiveClass) {
        return 'router-link'
      }
      return 'button'
    }
  },
}
</script>

<style lang='scss'>
  @mixin button-size($padding-y, $padding-x, $font-size, $line-height, $border-radius) {
    padding: $padding-y $padding-x;
    font-size: $font-size;
    line-height: $line-height;
    border-radius: $border-radius;
  }

  .va-button {
    display: inline-block;
    margin: 0.375rem 0.5rem;
    background-image: none;
    box-shadow: none;
    outline: none !important;
    border: 0;
    font-family: $font-family-sans-serif;
    text-transform: initial;
    cursor: pointer;
    transition: 0.3s cubic-bezier(0.25, 0.8, 0.5, 1);

    &--large {
      @include button-size(0.75rem, 2rem, 1.25rem, 1.3, 2rem);
      letter-spacing: 0.04rem;
      color: $white;

      &.va-button--with-left-icon {
        padding-left: 1.75rem;
      }

      &.va-button--with-right-icon {
        padding-right: 1.5rem;
      }
    }

    &--small {
      @include button-size(0.125rem, 1rem, 0.875rem, 1.43, 1rem);
      letter-spacing: 0.03rem;
      color: $white;

      &.va-button--with-left-icon, &.va-button--with-right-icon {
        padding-left: 0.5rem;
      }
    }

    &--normal {
      @include button-size(0.5rem, 1.5rem, 1rem, 1.5, 1.5rem);
      letter-spacing: 0.032rem;
      color: $white;

      &.va-button--with-left-icon {
        padding-left: 1rem;
      }

      &.va-button--with-right-icon {
        padding-right: 0.75rem;
      }
    }
  }

  $vuestic-colors: (
    success: (#63e5b3, #23e066, #77cea4, #40e583, #d6ffd3, #c0fbc7),
    danger: (#ff7455, #e34b4a, #b86e6d, #e34b4a, #ffebeb, #fbd2d2),
    warning: (#ffd72d, #feb900, #cbb06e, #ffc202, #fff3d1, #ffebb1),
    info: (#32b5e4, #2c82e0, #6c97ac, #2c82e0, #caeeff, #b2defb),
    gray: (#cdd0d5, #b4b6b9, #a3aab0, #babfc2, #e6e9ec, #dfe3e5),
    dark: (#576675, #34495e, #aebcca, #34495e, #afb6bb, #afb6bb)
  );

  @each $name, $colors in $vuestic-colors {
    $gradient-color1: nth($colors, 1);
    $gradient-color2: nth($colors, 2);
    $box-shadow: nth($colors, 3);
    $border-color: nth($colors, 4);
    $hover-color: nth($colors, 5);
    $focus-color: nth($colors, 6);

    .va-button--#{$name}.va-button--default{
      background-image: linear-gradient(to right, $gradient-color1, $gradient-color2);
      box-shadow: 0 0.125rem 0.19rem 0 $box-shadow;

      &:hover {
        background-image: linear-gradient(to right, lighten($gradient-color1, 15%), lighten($gradient-color2, 15%));
        color: $white !important;
      }

      &:active, &:focus {
        background-image: linear-gradient(to right, darken($gradient-color1, 15%), darken($gradient-color2, 15%));
        color: $white !important;
      }

      &.va-button--disabled {
        background-image: linear-gradient(to right, $gradient-color1, $gradient-color2);
        cursor: default;
        opacity: 0.4;
      }
    }

    .va-button--#{$name}.va-button--outline{
      background: transparent;
      border: solid 0.125rem $border-color;
      color: $border-color !important;

      &:hover {
        background-color: $hover-color;
        color: $border-color !important;
      }

      &:active {
        background-color: $focus-color;
        color: $border-color !important;
      }

      &.va-button--disabled {
        background: transparent;
        cursor: default;
        opacity: 0.4;
      }
    }

    .va-button--#{$name}.va-button--flat{
      background: transparent;
      border: 0 solid transparent;
      color: $border-color !important;

      &:hover {
        background-color: $hover-color;
        color: $border-color !important;
      }

      &:active, &:focus {
        background-color: $focus-color;
        color: $border-color !important;
        border: none;
      }

      &.va-button--disabled {
        background: transparent;
        cursor: default;
        opacity: 0.4;
      }
    }

    .va-button--#{$name}.va-button--default{
      i {
        color: $white;
      }
    }

    .va-button--#{$name}.va-button--outline, .va-button--#{$name}.va-button--flat {
      i {
        color: $border-color;
      }
    }
  }
</style>
