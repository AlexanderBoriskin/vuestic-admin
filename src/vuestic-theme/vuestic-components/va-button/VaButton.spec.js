import { shallowMount } from '@vue/test-utils'
import VaButton from './VaButton'

describe('VaButton', () => {
  /* 1. Button types */

  it('default button', () => {
    const wrapper = shallowMount(VaButton)

    expect(wrapper.find(VaButton).classes()).toContain('va-button--default')
  })

  it('outline button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { outline: true }
    })
    expect(wrapper.find(VaButton).classes()).toContain('va-button--outline')
  })

  it('outline button snapshot', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { outline: true }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('flat button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { flat: true }
    })
    expect(wrapper.find(VaButton).classes()).toContain('va-button--flat')
  })

  it('flat button snapshot', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { flat: true }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  /* 2. Button sizes */

  it('normal button', () => {
    const wrapper = shallowMount(VaButton)

    expect(wrapper.find(VaButton).classes()).toContain('va-button--normal')
  })

  it('large button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { large: true }
    })
    expect(wrapper.find(VaButton).classes()).toContain('va-button--large')
  })

  it('large button snapshot', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { large: true }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('small button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { small: true }
    })
    expect(wrapper.find(VaButton).classes()).toContain('va-button--small')
  })

  it('small button snapshot', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { small: true }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  /* 3. Button colors */

  it('info button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { color: 'info' }
    })
    expect(wrapper.find(VaButton).classes()).toContain('va-button--info')
  })

  it('info button snapshot', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { color: 'info' }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('danger button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { color: 'danger' }
    })

    expect(wrapper.find(VaButton).classes()).toContain('va-button--danger')
  })

  it('danger button snapshot', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { color: 'danger' }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('warning button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { color: 'warning' }
    })

    expect(wrapper.find(VaButton).classes()).toContain('va-button--warning')
  })

  it('warning button snapshot', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { color: 'warning' }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  /* 4. Button tags */

  it('default button', () => {
    const wrapper = shallowMount(VaButton)

    expect(wrapper.is('button')).toBe(true)
  })

  it('a button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        href: 'http://epic-spinners.epicmax.co/'
      }
    })
    expect(wrapper.is('a')).toBe(true)
  })

  it('a button 2 ', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        target: '_blank'
      }
    })
    expect(wrapper.is('a')).toBe(true)
  })

  it('a button snapshot', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        href: 'http://epic-spinners.epicmax.co/',
        target: '_blank'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('router-link button 1', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        to: { name: 'charts' }
      }
    })
    expect(wrapper.is('router-link')).toBe(true)
  })

  it('router-link button 2', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        to: { name: 'charts' },
        activeClass: 'va-button--active'
      }
    })
    expect(wrapper.is('router-link')).toBe(true)
  })

  it('router-link button snapshot', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        to: { name: 'charts' },
        append: true,
        activeClass: 'va-button--active'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  /* 5. Disabled buttons */

  it('disabled button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { disabled: true }
    })

    expect(wrapper.find(VaButton).classes()).toContain('va-button--disabled')
  })

  it('disabled snapshot', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { disabled: true }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  /* 6. Buttons with icons */

  it('icon left button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        icon: 'brandico brandico-facebook'
      }
    })
    expect(wrapper.find(VaButton).classes()).toContain('va-button--with-left-icon')
  })

  it('icon right button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        iconRight: 'iconicstroke iconicstroke-info'
      }
    })
    expect(wrapper.find(VaButton).classes()).toContain('va-button--with-right-icon')
  })

  it('both icon button snapshot', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        icon: 'brandico brandico-facebook',
        iconRight: 'iconicstroke iconicstroke-info'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
