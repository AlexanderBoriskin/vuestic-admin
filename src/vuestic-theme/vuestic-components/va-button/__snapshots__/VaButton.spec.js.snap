// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaButton a button 3 1`] = `
<a href="http://epic-spinners.epicmax.co/" target="_blank" class="va-button va-button--success va-button--default va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</a>
`;

exports[`VaButton a button snapshot 1`] = `
<a href="http://epic-spinners.epicmax.co/" target="_blank" class="va-button va-button--success va-button--default va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</a>
`;

exports[`VaButton both icon button snapshot 1`] = `
<button class="va-button va-button--success va-button--default va-button--without-title va-button--with-left-icon va-button--with-right-icon va-button--normal">
  <div class="va-button__content"><i class="va-button__content__icon va-button__content__icon-right brandico brandico-facebook"></i>
    <!----> <i class="va-button__content__icon va-button__content__icon-right iconicstroke iconicstroke-info"></i></div>
</button>
`;

exports[`VaButton danger button 2 1`] = `
<button class="va-button va-button--danger va-button--default va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton danger button snapshot 1`] = `
<button class="va-button va-button--danger va-button--default va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton disabled button 2 1`] = `
<button disabled="disabled" class="va-button va-button--success va-button--default va-button--disabled va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton disabled snapshot 1`] = `
<button disabled="disabled" class="va-button va-button--success va-button--default va-button--disabled va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton flat button 2 1`] = `
<button class="va-button va-button--success va-button--flat va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton flat button snapshot 1`] = `
<button class="va-button va-button--success va-button--flat va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton info button 2 1`] = `
<button class="va-button va-button--info va-button--default va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton info button snapshot 1`] = `
<button class="va-button va-button--info va-button--default va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton large button 2 1`] = `
<button class="va-button va-button--success va-button--default va-button--without-title va-button--large">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton large button snapshot 1`] = `
<button class="va-button va-button--success va-button--default va-button--without-title va-button--large">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton mixed icon 1`] = `
<button class="va-button va-button--success va-button--default va-button--without-title va-button--with-left-icon va-button--with-right-icon va-button--normal">
  <div class="va-button__content"><i class="va-button__content__icon va-button__content__icon-right brandico brandico-facebook"></i>
    <!----> <i class="va-button__content__icon va-button__content__icon-right iconicstroke iconicstroke-info"></i></div>
</button>
`;

exports[`VaButton outline button 2 1`] = `
<button class="va-button va-button--success va-button--outline va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton outline button snapshot 1`] = `
<button class="va-button va-button--success va-button--outline va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton router-link button snapshot 1`] = `
<router-link to="[object Object]" append="true" active-class="va-button--active" class="va-button va-button--success va-button--default va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</router-link>
`;

exports[`VaButton router-link snapshot 1`] = `
<router-link to="[object Object]" append="true" active-class="va-button--active" class="va-button va-button--success va-button--default va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</router-link>
`;

exports[`VaButton small button 2 1`] = `
<button class="va-button va-button--success va-button--default va-button--without-title va-button--small">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton small button snapshot 1`] = `
<button class="va-button va-button--success va-button--default va-button--without-title va-button--small">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton warning button 2 1`] = `
<button class="va-button va-button--warning va-button--default va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton warning button snapshot 1`] = `
<button class="va-button va-button--warning va-button--default va-button--without-title va-button--normal">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;
